api-admin/src/main/java/bio/terra/pearl/api/admin/service/forms/SurveyExtService.java

```diff
@@ -58,6 +64,59 @@ public Survey create(String portalShortcode, Survey survey, AdminUser adminUser)
     return surveyService.create(survey);
   }
 
+  public void delete(String portalShortcode, String surveyStableId, AdminUser adminUser) {
+    Portal portal = authUtilService.authUserToPortal(adminUser, portalShortcode);
+    List<Survey> existingVersions = surveyService.findByStableId(surveyStableId);
+    List<UUID> existingVersionIds = existingVersions.stream().map(BaseEntity::getId).toList();
+
+    if (existingVersionIds.size() == 0) {
+      throw new NotFoundException("Survey not found in portal " + portal.getName());
```

Comment:
You're not limiting your search to the portal (which is correct).
```suggestion
      throw new NotFoundException("Survey not found");
```
Also, you might as well move this check up a line -- just check the size of the survey list returned from the DB